hermes/bin/src/sign/certificate.rs

```diff
@@ -0,0 +1,110 @@
+//! x.509 certificate implementation.
+
+use std::{
+    fmt::Display,
+    path::{Path, PathBuf},
+};
+
+use x509_cert::der::DecodePem;
+
+use super::keys::PublicKey;
+
+/// Certificate file open and read error.
+#[derive(thiserror::Error, Debug)]
+pub(crate) struct CertificateFileError(PathBuf, Option<anyhow::Error>);
+impl Display for CertificateFileError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let msg = format!(
+            "Cannot open and read certificate file at {0}.",
+            self.0.display()
+        );
+        let err = self
+            .1
+            .as_ref()
+            .map(|msg| format!("{msg}"))
+            .unwrap_or_default();
+        writeln!(f, "{msg}\n{err}",)
+    }
+}
+
+/// Certificate decoding from string error.
+#[derive(thiserror::Error, Debug)]
+#[error("Cannot decode certificate from string. Invalid PEM format.")]
+pub(crate) struct CertificateDecodingError;
+
+/// x.509 cert instance.
+/// Wrapper over `x509_cert::Certificate`
+pub(crate) struct Certificate(x509_cert::Certificate);
+
+impl Certificate {
+    /// Create new certificate from file decoded in PEM format
+    pub(crate) fn from_file<P: AsRef<Path>>(path: P) -> anyhow::Result<Self> {
+        let str = std::fs::read_to_string(&path)
+            .map_err(|_| CertificateFileError(path.as_ref().into(), None))?;
+        Ok(Self::from_str(&str)
+            .map_err(|err| CertificateFileError(path.as_ref().into(), Some(err)))?)
+    }
+
+    /// Create new certificate from string decoded in PEM format
+    pub(crate) fn from_str(str: &str) -> anyhow::Result<Self> {
+        let cert = x509_cert::Certificate::from_pem(str.as_bytes())?;
+        Ok(Self(cert))
+    }
+
+    /// Get certificate's subject public key.
+    pub(crate) fn subject_public_key(&self) -> anyhow::Result<PublicKey> {
+        let subject_public_key = &self
+            .0
+            .tbs_certificate
+            .subject_public_key_info
+            .subject_public_key;
+
+        PublicKey::from_bytes(subject_public_key.raw_bytes())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use temp_dir::TempDir;
+
+    use super::*;
+
+    #[test]
+    fn certificate_from_file_test() {
```

Comment:
did you test parity between cryptoxide output and outputs after the introduction of dalek libs?